spec/px/service/kinesis/base_request_spec.rb

```diff
@@ -2,30 +2,33 @@
 
 describe Px::Service::Kinesis::BaseRequest do
 
-  subject { Px::Service::Kinesis::BaseRequest.new }
-  before(:each) do 
+  subject { Px::Service::Kinesis::BaseRequest.new.tap { |s| s.stream = "test" } }
+  let (:default_put_rate) { Px::Service::Kinesis::BaseRequest::DEFAULT_PUT_RATE }
+  let (:data) { {datakey: "value"} }
+
+  before :each  do
     Timecop.freeze
     stub_const("Px::Service::Kinesis::BaseRequest::FLUSH_LENGTH", 5)
-    subject.stream = "test"
   end
-  let (:default_put_rate) { Px::Service::Kinesis::BaseRequest::DEFAULT_PUT_RATE }
-  let(:data) { {datakey: "value"} }
-  
-  describe '#push_records' do
-    
-    context "when pushing data into buffer" do
-      before :each do
-        subject.push_records(data)
-      end
 
-      it "returns incremented buffer count" do
-        expect(subject.instance_variable_get(:@buffer).length).to eq(1)
-      end
+  describe '#queue_record' do
+    it "returns incremented buffer count" do
+      subject.queue_record(data)
+
+      expect(subject.buffer.size).to eq(1)
+    end
+
+    context "when enough time has elapsed" do
+      before :each do
+        subject
 
-      it "sets last send time" do
         Timecop.travel(15.seconds.from_now)
         Timecop.freeze
-        subject.flush_records
+      end
+
+      it "sets the last send time" do
+        subject.queue_record(data)
```

Comment:
Restructured these tests to better isolate setup code and tested code.  Also, since you're testing `queue_record` you should be calling `queue_record` and not `flush_records`.